FNX-5323 ⁃ [Bug] Sync isn't working anymore

We have a fairly recent regression, it seems. This used to work fine. I'm seeing the following in the logs while trying to sync:

```
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: java.util.concurrent.ExecutionException: java.lang.NoSuchMethodError: No virtual method sync(Lmozilla/appservices/places/SyncAuthInfo;)V in class Lmozilla/appservices/places/PlacesApi; or its super classes (declaration of 'mozilla.appservices.places.PlacesApi' appears in /data/app/org.mozilla.fenix-1/base.apk)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:4)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:48)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:1)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at java.lang.Thread.run(Thread.java:762)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: Caused by: java.lang.NoSuchMethodError: No virtual method sync(Lmozilla/appservices/places/SyncAuthInfo;)V in class Lmozilla/appservices/places/PlacesApi; or its super classes (declaration of 'mozilla.appservices.places.PlacesApi' appears in /data/app/org.mozilla.fenix-1/base.apk)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at mozilla.components.browser.storage.sync.RustPlacesConnection.sync(Connection.kt:2)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at mozilla.components.browser.storage.sync.PlacesStorage$sync$2.invokeSuspend(PlacesStorage.kt:10)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:2)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:13)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:1)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:1)
04-22 18:29:31.577  2790  2819 E WM-WorkerWrapper: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:42)
04-22 18:29:31.578  2790  2819 I WM-WorkerWrapper: Worker result FAILURE for Work [ id=fd62137e-591b-4321-850c-26967db59a3d, tags={ Common, Immediate, mozilla.components.feature.sync.WorkManagerSyncWorker } ]

```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -255,8 +255,6 @@ dependencies {
 
     implementation Deps.leanplum
 
-    implementation Deps.mozilla_places
-
     implementation Deps.mozilla_concept_engine
     implementation Deps.mozilla_concept_storage
     implementation Deps.mozilla_concept_toolbar

--- a/buildSrc/src/main/java/Dependencies.kt
+++ b/buildSrc/src/main/java/Dependencies.kt
@@ -26,7 +26,6 @@ private object Versions {
 
     const val appservices_gradle_plugin = "0.4.4"
     const val mozilla_android_components = "0.51.0-SNAPSHOT"
-    const val mozilla_appservices = "0.26.0"
 
     const val autodispose = "1.1.0"
     const val adjust = "4.11.4"
@@ -102,8 +101,6 @@ object Deps {
     const val mozilla_feature_findinpage = "org.mozilla.components:feature-findinpage:${Versions.mozilla_android_components}"
     const val mozilla_feature_session_bundling = "org.mozilla.components:feature-session-bundling:${Versions.mozilla_android_components}"
     const val mozilla_feature_site_permissions = "org.mozilla.components:feature-sitepermissions:${Versions.mozilla_android_components}"
-
-    const val mozilla_places = "org.mozilla.appservices:places:${Versions.mozilla_appservices}"
 
     const val mozilla_service_firefox_accounts = "org.mozilla.components:service-firefox-accounts:${Versions.mozilla_android_components}"
     const val mozilla_service_fretboard = "org.mozilla.components:service-fretboard:${Versions.mozilla_android_components}"

